scraper.py

```diff
@@ -114,38 +118,91 @@ def extract_score_data(elem):
     }
     return data
 
-if __name__ == '__main__':
+def generate_results(test=False, count=5, sort='highscore', reverse=False ):
     kwargs = {
         'Inspection_Start': '2/1/2013',
         'Inspection_End': '2/1/2015',
         'Zip_Code': '98109'
     }
-    if len(sys.argv) > 1 and sys.argv[1] == 'test':
+    if test:
         html, encoding = load_inspection_page()
     else:
         html, encoding = get_inspection_page(**kwargs)
-        #code to re-create files as needed
-        # html_part = open('inspection_page.html', 'w')
-        # html_part.write(html)
-        # html_part.closed
-        # encoding_part = open('inspection_page_encoding.html', 'w')
-        # encoding_part.write(encoding)
-        # encoding_part.closed
     doc = parse_source(html, encoding)
     listings = extract_data_listings(doc)
-    final_data = {}
-    for listing in listings:
+    data = []
+    for listing in listings[:count]:
```

Comment:
You are slicing the data before you sort it, which means that you won't be sorting *all* the records, only those within the slice you take.  Can you figure out how to sort the full set of results first, and only *then* slice to the desired count?